src/main/java/com/epam/reportportal/testng/TestNGService.java

```diff
@@ -518,19 +514,26 @@ private TestCaseIdEntry getTestCaseId(String codeRef, ITestResult testResult) {
 		TestCaseId testCaseId = getMethodAnnotation(TestCaseId.class, testResult);
 		return testCaseId != null ?
 				getTestCaseId(testCaseId, testResult) :
-				new TestCaseIdEntry(StringUtils.join(codeRef, Arrays.toString(testResult.getParameters())),
-						Arrays.deepHashCode(new Object[] { codeRef, testResult.getParameters() })
-				);
+				new TestCaseIdEntry(testCaseIdFromCodeRefAndParams(codeRef, testResult.getParameters()));
+	}
+
+	private String testCaseIdFromCodeRefAndParams(String codeRef, Object[] parameters) {
+		boolean isParametersPresent = Objects.nonNull(parameters) && parameters.length > 0;
+		return isParametersPresent ? codeRef + TRANSFORM_PARAMETERS.apply(parameters) : codeRef;
 	}
 
+	private static final Function<Object[], String> TRANSFORM_PARAMETERS = it -> "[" + Arrays.stream(it)
+			.map(param -> (String) param)
```

Comment:
String.valueOf() instead of type cast required to prevent type cast exceptions

Reply:
fixed